pages/forms/TripForm/index.js

```diff
@@ -0,0 +1,77 @@
+import useSWR from "swr";
+import CreateButton from "@/components/CreateButton";
+import CancelButton from "@/components/CancelButton";
+
+export default function TripForm() {
+  const { mutate } = useSWR("/api/trips");
+
+  async function handleAddTrip(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const tripData = Object.fromEntries(formData);
+
+    const newTrip = {
+      title: tripData.title,
+      location: [{ country: tripData.country, city: tripData.city }],
+      timePeriod: [
+        { startDate: tripData.startDate, endDate: tripData.endDate },
+      ],
+    };
+
+    const response = await fetch("/api/trips", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newTrip),
+    });
+
+    if (!response.ok) {
+      console.error(response.status);
+      return;
+    }
+
+    mutate();
+    event.target.reset();
+    window.location.href = "/";
```

Comment:
I suggest to use `router.push` to redirect the user back to the homepage as it is the preferred method in Next.js apps 🙂 
```suggestion
router.push("/");
```

Keep in mind to also import `useRouter` and to initialize the router object with `const router = useRouter();`